logstash-core/lib/logstash/plugins/event_factory_support.rb

```diff
@@ -0,0 +1,68 @@
+require 'logstash/util/thread_safe_attributes'
+
+module LogStash
+  module Plugins
+    module EventFactorySupport
+
+      extend LogStash::Util::ThreadSafeAttributes
+
+      # The event_factory method is effectively final and should not be re-defined by plugins.
+      #
+      # @return an event factory object with a `new_event(Hash)` API
+      # @since LS 7.14
+      lazy_init_attr(:event_factory, variable: :@_event_factory) { create_event_factory }
+
+      # The `targeted_event_factory` method is effectively final and should not be re-defined.
+      # If the plugin defines a `target => ...` option than this method will return a factory
+      # that respects the set target, otherwise (no target) returns {#event_factory}.
+      #
+      # @return an event factory object with a `new_event(Hash)` API
+      # @since LS 7.14
+      lazy_init_attr :targeted_event_factory, variable: :@_targeted_event_factory do
+        raise ArgumentError.new('config(:target) not present') unless respond_to?(:target)
+        target.nil? ? event_factory : TargetedEventFactory.new(event_factory, target)
+      end
+
+      private
+
+      # @api private
+      # @since LS 7.14
+      def create_event_factory
```

Comment:
do we drop this method for now? to align with the [support mixin](https://github.com/logstash-plugins/logstash-mixin-event_support/pull/2#discussion_r657261492)

Reply:
It's private, so not necessarily something we need to align on. It is also a good hook point for future efforts in LS core that don't need backport support (things like alternate `Event` implementation experiments like copy-on-write or field reference parser changes), so I expect even if we remove it now we will add it back in short order.

Reply:
okay, sticking around just wanted to double check given what `private` means in Ruby land :wink: